app/models/form.rb

```diff
@@ -175,9 +175,10 @@ def visible_questionings
     questionings.reject{|q| q.hidden?}
   end
 
-  # returns questionings that work with sms forms and are not hidden
+  # returns hash of questionings that work with sms forms and are not hidden
   def smsable_questionings
-    questionings.reject{|q| q.hidden? || !q.question.qtype.smsable?}
+    smsable_questionings = questionings.reject{|q| q.hidden? || !q.question.qtype.smsable?}
+    smsable_questionings.index_by.with_index { |_, i| i + 1 }
```

Comment:
`with_index` is a method on Enumerator which essentially adds an index to an existing enumerator, so `['cat', 'dog', 'bunny'].map.with_index{ |animal, index| "#{index}:#{animal}" }` would give you `["0:cat", "1:dog", "2:bunny"]`

`index_by` is more complicated than `map`, and ruby is doing some weird stuff behind the scenes to make Enumerators work that I don't *fully* understand because the entire class is written in raw C, but there's basically some kind of delegation back to the original Enumerator happening, and `with_index` is just some sort of a wrapper.